pkg/middleware/recovery/recovery.go

```diff
@@ -0,0 +1,48 @@
+package recovery
+
+import (
+	"context"
+	"net/http"
+
+	grpc_recovery "github.com/grpc-ecosystem/go-grpc-middleware/recovery"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/openfga/openfga/pkg/logger"
+)
+
+func Panic(next http.Handler, logger logger.Logger) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer func() {
+			if err := recover(); err != nil {
+				logger.Error("Recovered from panic",
+					zap.Any("panic_info", err))
+				http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+			}
+		}()
+		next.ServeHTTP(w, r)
+	})
+}
+
+// UnaryPanicInterceptor recovers from panics in unary handlers
+func UnaryPanicInterceptor(logger logger.Logger) grpc.UnaryServerInterceptor {
```

Comment:
We actually dont' need this and the streaming interceptor below if you do what I mentioned above.

```go
func PanicRecoveryHandler(logger Logger) recovery.RecoveryHandlerFuncContext {...}
```
and then, in `run.go`, for example:
```go
grpc_recovery.UnaryServerInterceptor(  
    grpc_recovery.WithRecoveryHandlerContext(  
        recovery.PanicRecoveryHandler(logger),  
    ),  
),  

...

grpc_recovery.StreamServerInterceptor(
    grpc_recovery.WithRecoveryHandler(
        recovery.PanicRecoveryHandler(logger),
    ),
),
```

Reply:
Done